src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
I agree that `true` doesn't always imply `successful`, but it's a pretty widely used convention and the documentation can be changed to make it explicit.